internal/controller/promotionstrategy_controller.go

```diff
@@ -477,3 +406,98 @@ func (r *PromotionStrategyReconciler) copyCommitStatuses(ctx context.Context, cs
 
 	return nil
 }
+
+func (r *PromotionStrategyReconciler) tryToMergePullRequest(ctx context.Context, ps *promoterv1alpha1.PromotionStrategy, proposedCommitMap map[string]*promoterv1alpha1.ProposedCommit) error {
```

Comment:
Why prefix with `try` and not just `mergePullRequest`? The caller should be the one with the responsibility to know how to react on errors.

Reply:
Yea the function is not actually retrying, the function name was in regards to trying to describe the action of trying to merge pull requests if the commit status checks where green.